src/main/java/com/oracle/nosql/spring/data/core/NosqlTemplateBase.java

```diff
@@ -178,6 +188,190 @@ protected boolean doCreateTableIfNotExists(
         return tableState == TableResult.State.ACTIVE;
     }
 
+    protected boolean doCheckExistingTable(NosqlEntityInformation<?, ?> entityInformation) {
+        final String colField = "fields";
+        final String colNameField = "name";
+        final String colTypeField = "type";
+        final String shardField = "shardKey";
+        final String primaryField = "primaryKey";
+        final String ttlField = "ttl";
+        final String identityField = "identity";
+        final String schemaVersionField = "json_version";
+        final int supportedSchemaVersion = 1;
+
+        List<String> errors = new ArrayList<>();
+        try {
+            TableResult tableResult = doGetTable(entityInformation);
+
+            // table does not exist return false
+            if (tableResult == null) {
+                return false;
+            }
+
+            /* If table already exist in the database compare and throw error if
+               mismatch*/
+            MapValue tableSchema = JsonUtils.createValueFromJson(
+                            tableResult.getSchema(),
+                            new JsonOptions().setMaintainInsertionOrder(true)).
+                    asMap();
+
+            int currentSchemaVersion = tableSchema.get(schemaVersionField).
```

Comment:
I mentioned this to Cezar -- do not use the version for this sort of comparison; even if it changes the information you need should not

Reply:
Removed JSON schema version check. Now code throws IllegalStateException when failed to parse JSON schema.